_rules/required-form-fields-not-completed-334972.md

```diff
@@ -0,0 +1,243 @@
+
+## Expectation 1
+
+After triggering the submission of the [form element][] that is an [ancestor][] of the target element, each target element that was not [interacted with][] is identified by a text message.
+
+**Note**: A text message may identify an element in several ways, including:
+
+- By referring to its label or accessible name
+- By being visually placed in the vicinity of the element
+
+## Expectation 2
+
+The content of the text message is [visible][], [included in the accessibility tree][] and indicates that the target element is [required][].
+
+## Assumptions
+
+_There are currently no assumptions_
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 3.3.1: Error Identification](https://www.w3.org/WAI/WCAG21/Understanding/error-identification)
+- [G83: Providing text descriptions to identify required fields that were not completed](https://www.w3.org/WAI/WCAG21/Techniques/general/G83)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The error message that shows near the `input` element when it is not filled indicates the field must be filled.
+
+```html
+<script>
+	function processForm() {
+		if (document.getElementById('text_field').value.length === 0) {
+			document.getElementById('error').innerText = 'You must fill the name field'
+		}
+	}
+</script>
+
+<form>
+	<label for="text_field">Name (required)</label>
+	<input type="text" id="text_field" required />
+	<input type="button" value="Submit" onclick="processForm()" />
+	<div id="error"></div>
+</form>
+```
+
+#### Passed Example 2
+
+The error message identifies any required `input` element that has not been filled. The error message refers to the `input` elements by their labels.
+
+```html
+<script>
+	function processForm() {
+		document.getElementById('error').innerText = ''
+		if (document.getElementById('name').value.length === 0) {
+			document.getElementById('error').innerText += 'You must fill the name field. '
+		}
+		var color = document.forms[0].color.value
+		if (color.length === 0) {
+			document.getElementById('error').innerText += 'You must pick a color.'
+		}
+	}
+</script>
+
+<form>
+	<h2 id="error"></h2>
+	<label for="name">Name (required)</label>
+	<input type="text" id="name" required />
+	<br />
+	<label for="address">Address</label>
+	<input type="text" id="address" />
+	<p>Pick a color (required)</p>
+	<label><input type="radio" name="color" value="blue" required />Blue</label>
+	<label><input type="radio" name="color" value="yellow" />Yellow</label>
+	<br />
+	<input type="button" value="Submit" onclick="processForm()" />
+</form>
+```
+
+### Failed
+
+#### Failed Example 1
+
+No error message is provided.
+
+```html
+<form>
+	<label for="text_field">Name (required)</label>
+	<input type="text" id="text_field" required />
+	<input type="button" value="Submit" />
```

Comment:
This is not a submit button. The expectation calls for actually submitting the form. If you actually were to use a submit button, an error message would be shown. None of these failed cases seem valid to me.

Reply:
The expectation has been reworded according to your suggestion